samples/ziti-enroller/src/main/kotlin/org/openziti/ZitiEnroller.kt

```diff
@@ -48,6 +60,80 @@ object ZitiEnroller {
         }
     }
 
+    private class verify: CliktCommand(help = "verify identity file. OTP will be requested if identity is enrolled in MFA") {
+        val idFile by option(help = "identity configuration file.").file().required().validate {
+            it.exists()
+        }
+
+        override fun run() {
+            val ztx = Ziti.newContext(idFile, charArrayOf(), object : Ziti.AuthHandler{
+                override fun getCode(ztx: ZitiContext, mfaType: MFAType, provider: String) =
+                    CompletableFuture.supplyAsync {
+                        print("Enter MFA code for $mfaType/$provider[${ztx.getId()?.name}]: ")
+                        val code = readLine()
+                        code!!
+                    }
+            })
+
+            val j = GlobalScope.launch  {
+                ztx.statusUpdates().collect {
+                    println("status: $it")
+                    when(it) {
+                        ZitiContext.Status.Loading, ZitiContext.Status.Authenticating -> {}
+                        ZitiContext.Status.Active -> {
+                            println("verification success!")
+                            cancel()
+                        }
+                        is ZitiContext.Status.NotAuthorized -> {
+                            cancel("verification failed!", it.ex)
+                        }
+                        else -> cancel("unexpected status")
+                    }
+                }
+            }
+
+            runBlocking {
+                j.join()
+                ztx.destroy()
+            }
+
+        }
+    }
+
+    private class mfa: CliktCommand(help = "Enroll identity in MFA") {
+        val idFile by option(help = "identity configuration file.").file().required().validate {
+            it.exists()
+        }
+
+        override fun run() {
+            val ztx = Ziti.newContext(idFile, charArrayOf())
+
+            val j = GlobalScope.launch {
+                ztx.statusUpdates().takeWhile { it != ZitiContext.Status.Active }.collectLatest { println(it) }
+                val mfa = ztx.enrollMFA()
+                println(mfa)
+
+                print("Enter OTP code: ")
+                val code = readLine()
+                ztx.verifyMFA(code!!.trim())
+            }
+            runBlocking { j.join() }
+
+            ztx.destroy()
+        }
+
+    }
+
+    private class Cli : CliktCommand(name = "ziti-enroller") {
+
+        init {
+            subcommands(enroll(), verify(), mfa())
+        }
+
+        override fun run() {
+        }
+    }
+
```

Comment:
I saw your todo for "reset". For clarity, in this sample I don't see:

- re-fetch verification codes (deleted/disposed of them on accident)
- generate new verification codes (used too many or physically lost control of existing verification codes)
- remove MFA

Reply:
I also don't see the analogous functions necessary for those actions in `ziti/src/main/kotlin/org/openziti/api/Controller.kt`

Reply:
working on it ;)